### PR TITLE
Formatter numbers with thousand separator 

### PR DESCRIPTION
**Pull request by vadymmarkov**
# Why?

Because big numbers look better with separator.

# What?

- Add static number formatter to use across the app
- Format numbers with thousand separator everywhere except for years

# Show me

### Before

before

### After

after

### CODE REVIEW

File: Sources/Charcoal/Extensions/NumberFormatterExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import Foundation
+
+extension NumberFormatter {
+    static let formatterWithSeparator: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
```
Comment (shredlocker) [inline Sources/Charcoal/Extensions/NumberFormatterExtensions.swift]: I think we should use `.decimal`. I think the only difference is `.currency` has two decimals.

File: Sources/Charcoal/Filters/Root/RootFilterViewController.swift [hunk 2]
```diff
@@ -111,6 +111,12 @@ final class RootFilterViewController: FilterViewController {
         ])
     }
 
+    private func updateBottomButtonTitle() {
+        let localizedString = String(format: "showResultsButton".localized(), filter.numberOfResults)
+        let title = localizedString.replacingOccurrences(of: "\(filter.numberOfResults)", with: filter.formattedNumberOfResults)
```
Comment (shredlocker) [inline Sources/Charcoal/Filters/Root/RootFilterViewController.swift]: Nice 💯

File: Sources/Charcoal/Extensions/NumberFormatterExtensions.swift (hunk 1 above)
Comment (vadymmarkov) [reply Sources/Charcoal/Extensions/NumberFormatterExtensions.swift]: Good point! Fixed.

File: Sources/Charcoal/Filters/Root/RootFilterViewController.swift (hunk 2 above)
Comment (vadymmarkov) [reply Sources/Charcoal/Filters/Root/RootFilterViewController.swift]: That's because `String(format:` removes whitespaces 🙈

**Pull request merged**